(GH-2303) Add module-install config option (#2325)
puppetlabs/bolt: documentation/bolt_known_issues.md

# Known issues

## Resolving module dependencies does not support proxies or alternate Forge

When running the `bolt module add|install` commands or `Add|Install-BoltModule`
cmdlets, Bolt does not support a configured proxy or alternate Forge when it
resolves module dependencies, even if the `module-install` configuration
option is set. Support for resolving module dependencies with proxies or an
alternate Forge requires changes to one of Bolt's gem dependencies, which is
currently in progress.

While Bolt will not resolve module dependencies with proxies or an alternate
Forge, it will respect this configuration when installing modules.

If your project configures the `module-install` option, you may experience
one of the following issues:

- On a restricted network where a proxy is required, resolving modules may
  cause Bolt to fail.

- When an alternate Forge is specified, Bolt may resolve and install a
  different set of modules than expected.

### Install modules without resolving dependencies

To avoid this limitation and any potential errors, you can manually edit your

> Puppetfile and install modules without resolving dependencies. This is a similar

Reviewer: ```suggestion
Puppetfile to add and install modules without resolving dependencies. This is a similar
```

> similar workflow as the `bolt puppetfile install` and

Reviewer 2: ```suggestion
similar workflow to using the `bolt puppetfile install` and
```

`Install-BoltPuppetfileModules` commands.

For example, if your project requires the `puppetlabs/apache` module, but
you need to download modules hosted on an alternate Forge, you should write

> your Puppetfile manually. That Puppetfile may look similar to this:

Reviewer 2: ```suggestion
your Puppetfile manually. That Puppetfile might look similar to this:
```
Just a style thing - we prefer 'might' over 'may'
